lib/WorkDoneProgress/ClientCapabilityDependentProgressNotifier.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Phpactor\LanguageServer\WorkDoneProgress;
+
+use Amp\Promise;
+use Phpactor\LanguageServerProtocol\ClientCapabilities;
+use Phpactor\LanguageServer\Core\Server\ClientApi;
+
+final class ClientCapabilityDependentProgressNotifier implements ProgressNotifier
+{
+    /**
+     * @var ProgressNotifier
+     */
+    private $notifier;
+
+    public function __construct(ClientApi $api, ClientCapabilities $capabilities)
+    {
+        if ($capabilities->window['workDoneProgress'] ?? false) {
+            $this->notifier = new WorkDoneProgressNotifier($api);
+        } else {
+            $this->notifier = new MessageProgressNotifier($api);
+        }
```

Comment:
not  blocker: would remove `else` here and delegate to `createNotifier()` with an early return? (mostly because `else` is avoided just about everywhere else)